fedora/clients/fasjson.py

```diff
@@ -101,3 +101,51 @@ async def get_users_by_matrix_id(self, matrix_id: str) -> dict | str:
             )
 
         return searchresult[0]
+
+    async def get_user_groups(self, username, params=None):
+        """
+        Retrieves all groups a user belongs to, including membership type (member or sponsor).
+
+        Args:
+            username (str): Username of the user to query.
+            params (dict, optional): Additional query parameters to include in the request.
+
+        Returns:
+            List[dict] or None:
+                - A list of dictionaries containing group information:
+                    - groupname (str): Name of the group.
+                    - membership_type (str): "member" or "sponsor" depending on the user's role.
+                - None (if the user is not found or there's an error).
+
+        Raises:
+            InfoGatherError: If there's an error fetching data from Fasjson or if the user is not found (404).
+        """
+
+        try:
+            response = await self._get("/".join(["users", username, "groups"]), params=params)
+            user_groups = response.json().get("groups", [])
+
+            group_details = []
+            for groupname in user_groups:
+                # Fix - i realized at some point that get group membership, just returns members/sponsors in a group
+                sponsors = await self.get_group_membership(groupname, "sponsors", params=params)
+
+            if sponsors is not None:
+                if username in sponsors:
+                    membership_type = "sponsor"
+                else:
+                    membership_type = "member"
+            else:
+                # Handle the case when sponsors is None (no sponsors for the group)
+                membership_type = "member"
+
+            group_details.append({"groupname": groupname, "membership_type": membership_type})
+
+            return group_details
+
+        except NoResult as e:
+            raise InfoGatherError(
+                f"Sorry, but Fedora Accounts user '{username}' does not exist"
+            ) from e
+        except InfoGatherError as e:
+            raise
```

Comment:
My Python is a little rusty, but I _think_ this exception handler is equivalent to not having the except block at all, right?